### ISSUE TITLE
How to hide users in search results

### ISSUE THREAD

**Issue by dejalab** [RocketChat/Rocket.Chat#10162]
Anyone know if is possible to hide users by role or by other way in the search results?
We need to implement a solution in our college to let anyone to find teachers (to talk to), but not other students. 

Many thanks for your help

JR*

**Comment by rasos** [comment]
Yes, having a role to be listed in search results seems important to any data protection sensible environment. In the European Union, a new regulation will be valid from 25th May 2018 which requires providers not to publish any personal data.

**Comment by kaiiiiiiiii** [comment]
Hi @dejalab ,

Rocket.Chat uses Github mainly for discussing bugs though.

I would therefore like to point you to some places where you can try asking for help or suggest feature requests. Usually, some friendly human being has time for you and your request:

The Rocket.Chat Community Support Channel: https://open.rocket.chat/channel/support
The Rocket.Chat Community Support Forum: https://forums.rocket.chat/c/community-support

Thanks,
Kai

**Comment by engelgabriel** [comment]
We could have a setting on the admin, for managers to define an object to be merged with the DB query object before the query is executed. This could be very simple to implement.
